devo/sender/data.py

```diff
@@ -63,12 +70,164 @@ def __init__(self, address=None, key=None, cert=None, chain=None,
+            chain_raw = chain_file.read()
+            certificate_obj = crypto.load_certificate(
+                crypto.FILETYPE_PEM, certificate_raw)
+            certificates_chain = crypto.X509Store()
+            for certificate in pem.parse(chain_raw):
+                certificates_chain.add_cert(
+                    crypto.load_certificate(
+                        crypto.FILETYPE_PEM, str(certificate)))
+            store_ctx = crypto.X509StoreContext(
+                certificates_chain, certificate_obj)
+        try:
+            store_ctx.verify_certificate()
+        except crypto.X509StoreContextError as message:
+            raise DevoSenderException(
+                "Error in config, the chain: " + self.chain +
+                " is not compatible with the certificate: " + self.cert +
+                "\noriginal error: " + str(message))
+        return True
+
+    def check_config_certificate_address(self):
+        """
+        Check if the certificate is compatible with the
+        address, also check is the address and port are
+        valid.
+
+        :return: Boolean true or raises an exception
+        """
+        sock = socket.socket()
+        context = SSL.Context(SSL.TLS_CLIENT_METHOD)
+        sock.settimeout(10)
+        connection = SSL.Connection(context, sock)
+        try:
+            connection.connect((self.address[0], self.address[1]))
+        except socket.timeout as message:
+            raise DevoSenderException(
+                "Possible error in config, a timeout could be related " +
+                "to an incorrect address/port: " + str(self.address) +
+                "\noriginal error: " + str(message))
+        except ConnectionRefusedError as message:
+            raise DevoSenderException(
+                "Error in config, incorrect address/port: "
+                + str(self.address) +
+                "\noriginal error: " + str(message))
+        sock.setblocking(True)
+        connection.do_handshake()
+        server_chain = connection.get_peer_cert_chain()
+
+        with open(self.chain, "rb") as chain_file:
+            chain = chain_file.read()
+            chain_certs = []
+            for _ca in pem.parse(chain):
+                chain_certs.append(
+                    crypto.load_certificate(crypto.FILETYPE_PEM, str(_ca)))
+
+        server_common_names = {}
+        client_common_names = {}
+        for temp_cert in server_chain:
+            for key, value in temp_cert.get_subject().get_components():
+                if key.decode("utf-8") == "CN":
+                    server_common_names[value] = ""
+
+        for temp_cert in chain_certs:
+            for key, value in temp_cert.get_issuer().get_components():
+                if key.decode("utf-8") == "CN":
+                    client_common_names[value] = ""
+
+        for common_name in client_common_names:
+            if common_name in server_common_names:
+                return True
```

Comment:
**suggestion:** I believe this code is referred to my previously-asked questions about the "fixed indexes" usage. This now looks much better.

I, however, do not understand the usage of dictionaries if you actually do not store any particular value (only empty strings) to then check if both dicts share one same value. I would transform this code in the following (not tested):

```suggestion
        def get_common_names(certs_iter, components_attr):
            return {
                value
                for temp_crt in certs_iter
                # We dinamically call the temp_cert's method whose name is 
                # components_attr to get either the subject, the issuer...
                # From it, we can get their components.
                for key, value in getattr(temp_crt, components_attr)().get_components()
                if key.decode("utf-8") == "CN"
            }

        server_common_names = get_common_names(server_chain, "get_subject")
        client_common_names = get_common_names(chain_certs, "get_issuer")
        if server_common_names & client_common_names:
            # the sets intersect => the client and server have a CN in common
            return True
```

> Please note it would be better if you move the defined function somewhere else.